antenna/configlib.py

```diff
@@ -0,0 +1,332 @@
+"""This module contains the configuration infrastructure allowing for deriving
+configuration from a .ini file and the process environment.
+
+In order of precedence:
+
+1. settings overrides (usually only applies when running tests)
+2. process environment
+3. .ini file (specified by ANTENNA_INI environment variable)
+
+Example of usage::
+
+    from antenna.configlib import config
+
+    DEBUG = config('DEBUG', default='True', parser=bool)
```

Comment:
So, stepping back a bit and talking about "icky", "pythonic" and "feels"... these are words that imply subjective measures. That's cool. My experience has been that people use words like this when they can't argue in a more tangible way the pros and cons of a decision. In this case, I think this is Pythonic because it it creates an API that's explicit and simple in nature.

This API requires the programmer to explicitly specify the parser. That nixes a class of possible bugs and it's less code to implement and test. I don't think the tradeoff for adding code to infer the parser from the default here is good enough.

Regarding the default, I think making it always a string makes it consistent. It's *always* the string value that you would pass to the parser, no guessing required. Plus reading the code illuminates what kinds of values you can use in your ini file and os environment.

That seems like a good tradeoff to me for a configuration library that's trying to be the smallest one possible.